packages/common/src/dotlottie-player.ts

```diff
@@ -683,16 +943,95 @@ export class DotLottiePlayer {
     return this._currentAnimationId;
   }
 
+  public get activeStateId(): string | undefined {
+    return this._activeStateId;
+  }
+
+  /**
+   * Gets the state machines from file and starts the specified state machine.
+   * @param stateId - The identifier of the state machine to use.
+   * @returns
+   */
+  protected async _startInteractivity(stateId: string): Promise<void> {
+    if (!this._inInteractiveMode) {
+      logError(
+        "Can't start initeractivity. Not in interactive mode. Call enterInteractiveMode(stateId: string) to start.",
+      );
+
+      return;
+    }
+    if (this._dotLottieLoader.stateMachinesMap.size === 0) {
+      await this._dotLottieLoader.getStateMachines();
+    }
+
+    if (this._dotLottieLoader.stateMachinesMap.size === 0) {
+      throw createError('No interactivity states are available.');
+    }
+
+    if (stateId === 'undefined') {
+      throw createError('stateId is not specified.');
+    }
+
+    if (!this._stateMachine) {
+      this._stateMachine = new DotLottieStateMachine(Array.from(this._dotLottieLoader.stateMachinesMap.values()), this);
```

Comment:
IMO passing an instance of DotLottiePlayer to DotLottieStateMachine creates a circular dependency between objects and i think we should go with decoupling as much as possible and logically DotLottiePlayer only depend on DotLottieStateMachine according to the initial modular design of plugins/modules idea ? 

maybe to enable communication between the two instances, it might be a good idea to expose some sort of API interface for such communication from the DotLottieStateMachine